### PR TITLE
Fix Stroke Bug for 0 linewidth

### PR DESCRIPTION
**Pull request by madhur-tandon**

### CODE REVIEW

File: packages/matplotlib/src/html5_canvas_backend.py [hunk 1]
```diff
@@ -214,6 +215,8 @@ def set_joinstyle(self, js):
             raise ValueError('Unrecognized join style. Found {0}'.format(js))
 
     def set_linewidth(self, w):
+        if w == 0:
+            self.stroke = False
```
Comment (mdboom) [inline packages/matplotlib/src/html5_canvas_backend.py]: We also need to make sure stroke is `True` if `w != 0`.  So maybe just:

```
self.stroke = (w != 0)
```

File: packages/matplotlib/src/html5_canvas_backend.py [hunk 2]
```diff
@@ -308,12 +311,12 @@ def draw_path(self, gc, path, transform, rgbFace=None):
...
         transform += Affine2D().scale(1, -1).translate(0, self.height)
         self._path_helper(self.ctx, path, transform, figure_clip)
+        if gc.stroke:
+            self.ctx.stroke()
 
         if rgbFace is not None:
             self.ctx.fill()
             self.ctx.fillStyle = '#000000'
-        else:
-            self.ctx.stroke()
```
Comment (mdboom) [inline packages/matplotlib/src/html5_canvas_backend.py]: I think the stroke should always be drawn after the fill, so move it down here.

File: packages/matplotlib/src/html5_canvas_backend.py (hunk 1 above)
Comment (madhur-tandon) [reply packages/matplotlib/src/html5_canvas_backend.py]: Aah yes, doing, thanks!

File: packages/matplotlib/src/html5_canvas_backend.py (hunk 2 above)
Comment (madhur-tandon) [reply packages/matplotlib/src/html5_canvas_backend.py]: On it :)

**Comment by madhur-tandon** [comment]
@mdboom Prior to changing `self.stroke = (w != 0)`, all the tests passed in circleCI. This change doesn't affect my local testing environment, but somehow, the tests fail in `circleCI` with non-zero deviation. Should I revert back since the default value of `self.stroke` is anyways `True` (initialized in the constructor) ?

**Comment by mdboom** [comment]
> Should I revert back since the default value of self.stroke is anyways True (initialized in the constructor) ?

But it's possible matplotlib might draw things with `linewidth == 0` and then later `linewidth == 1`, and that means those wouldn't be drawn.  Perhaps we require an update of the baseline images again?

**Comment by madhur-tandon** [comment]
Yes, I understand. But `git` doesn't show any changes to the newly saved images. I will probably use `ssh` now to check what's the issue.

**Comment by madhur-tandon** [comment]
@mdboom For some reason, `git` didn't show me before that the newly generated reference images are different. Nevertheless, I have uploaded the new ones. All tests related to `rendering` pass now. This is ready :)

**Pull request merged**